test/celeritas/ext/detail/TrackProcessor.test.cc

```diff
@@ -0,0 +1,349 @@
+#include <G4ParticleDefinition.hh>
+#include <G4ParticleTable.hh>
+#include <G4ProcessType.hh>
+#include <G4Step.hh>
+#include <G4StepPoint.hh>
+#include <G4Track.hh>
+#include <G4VProcess.hh>
+#include <G4VUserTrackInformation.hh>
+
+#include "corecel/Types.hh"
+#include "celeritas/SimpleCmsTestBase.hh"
+#include "celeritas/phys/PDGNumber.hh"
+
+#include "celeritas_test.hh"
+
+namespace celeritas
+{
+namespace detail
+{
+namespace test
+{
+//---------------------------------------------------------------------------//
+
+class MockUserTrackInformation : public G4VUserTrackInformation
+{
+  public:
+    explicit MockUserTrackInformation(int value) : value_(value) {}
+    int value() const { return value_; }
+
+  private:
+    int value_;
+};
+
+// Simple mock pointer class to test process pointer storage/restoration
+class MockProcessPointer
+{
+  public:
+    explicit MockProcessPointer(std::string name) : name_(std::move(name)) {}
+    std::string const& GetTestName() const { return name_; }
+
+  private:
+    std::string name_;
+};
+
+//---------------------------------------------------------------------------//
+
+class TrackProcessorTest : public ::celeritas::test::SimpleCmsTestBase
+{
+  protected:
+    using VecParticle = TrackProcessor::VecParticle;
+    using size_type = ::celeritas::size_type;
+
+    VecParticle make_particles()
+    {
+        // Load particles from Geant4
+        this->physics();
+
+        VecParticle result;
+        auto& table = *G4ParticleTable::GetParticleTable();
+        for (auto p : {pdg::gamma(), pdg::electron(), pdg::positron()})
+        {
+            result.push_back(table.FindParticle(p.get()));
+        }
+        return result;
+    }
+};
+
+//---------------------------------------------------------------------------//
+
+TEST_F(TrackProcessorTest, construction)
+{
+    // Create an empty processor first to test basic construction
+    TrackProcessor::VecParticle empty_particles;
+    TrackProcessor processor(empty_particles);
+
+    // Test that end_event works
+    processor.end_event();
+}
+
+//---------------------------------------------------------------------------//
+
+TEST_F(TrackProcessorTest, primary_registration)
+{
+    auto particles = make_particles();
+    TrackProcessor processor(particles);
+
+    // Create a primary track
+    auto primary_track = std::make_unique<G4Track>(
+        new G4DynamicParticle(particles[0], G4ThreeVector(1, 0, 0)),
+        0.0,
+        G4ThreeVector(0, 0, 0));
+    primary_track->SetTrackID(123);
+    primary_track->SetParentID(0);
+
+    // Add user information
+    auto user_info = std::make_unique<MockUserTrackInformation>(42);
+    primary_track->SetUserInformation(user_info.release());
+
+    // Set creator process using mock process pointer
+    auto mock_process = std::make_unique<MockProcessPointer>("TestCompton");
+    auto* process_ptr = reinterpret_cast<G4VProcess*>(mock_process.get());
+    primary_track->SetCreatorProcess(process_ptr);
```

Comment:
This looks illegal. Can't you just create a minimal G4VProcess ?

Reply:
I've added a minimal processor `reinterpret_cast` between pointer to unrelated types is legal, but accessing members is UB.

Reply:
Seems that there is an issue linking G4VProcess:
```
/usr/bin/ld: test/celeritas/CMakeFiles/celeritas_ext_detail_TrackProcessor.dir/ext/detail/TrackProcessor.test.cc.o: undefined reference to symbol '_ZN10G4VProcess16SetMasterProcessEPS_'
/usr/bin/ld: /opt/view/lib64/libG4processes.so: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
```